vkcs/compute/resource_instance.go

```diff
@@ -405,6 +405,40 @@ func ResourceComputeInstance() *schema.Resource {
 	}
 }
 
+func resourceComputeInstanceCreateWithWarnings(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	diags := resourceComputeInstanceCreate(ctx, d, meta)
+	if diags.HasError() {
+		return diags
+	}
+
+	// check synchronization of fields `delete_on_termination` and `source_type`
+	if vL, ok := d.GetOk("block_device"); ok {
+		for _, bd := range vL.([]interface{}) {
+			bdM := bd.(map[string]interface{})
+			deleteOnTermination := bdM["delete_on_termination"].(bool)
+			sourceType := bdM["source_type"].(string)
+			switch {
+			case sourceType == "blank" || sourceType == "image" || sourceType == "snapshot":
```

Comment:
This will lead to adding several duplicate diagnostics in some cases, it's better to `break` the loop here.

Reply:
done